flagsmith/analytics.py

```diff
@@ -29,9 +38,11 @@ def __init__(self, environment_key: str, base_api_url: str, timeout: int = 3):
         """
         self.analytics_endpoint = base_api_url + ANALYTICS_ENDPOINT
         self.environment_key = environment_key
-        self._last_flushed = datetime.now()
         self.analytics_data = {}
         self.timeout = timeout
+        self.session = session or FuturesSession(max_workers=4)
```

Comment:
Are we letting user pass request.session here, instead of hard-coded `FutureSession`?